### PR TITLE
feat(docs): add automatically generate a link to repository in docs if available

### PR DESCRIPTION
**Pull request by mrgleam**
The PR add automatically generate a link to repository in docs if available for issue #858

Screen Shot 2564-01-19 at 20 23 01

Screen Shot 2564-01-19 at 20 22 49

### CODE REVIEW

File: src/docs.rs [hunk 1]
```diff
@@ -88,6 +88,30 @@ pub fn generate_html(
...
+            path: url.clone(),
+        }),
+        Repository::None => None,
+    }
+    .into_iter()
+    .collect::<Vec<Link>>();
+
+    let links = &doc_links
+        .iter()
+        .chain(repo_link.iter())
```
Comment (lpil) [inline src/docs.rs:109]: I think if we call `into_iter` rather than `iter` we remove the `map` call that clones the link,

**Review by lpil — COMMENTED** [review]
Brilliant, thank you! I've left one inline note.

File: src/docs.rs (hunk 1 above)
Comment (mrgleam) [reply src/docs.rs:109]: I try to change line 108 from ` iter()` to` into_iter()` and remove the `map`

I got error `a value of type std::vec::Vec<docs::Link> cannot be built from an iterator over elements of type &docs::Link`

Can I use `cloned()` for remove the `map`? 

```
    let links = &doc_links
        .iter()
        .chain(repo_link.iter())
        .cloned()
        .collect::<Vec<Link>>()
```

Comment (lpil) [reply src/docs.rs:109]: Sounds good! I was surprised that it didn't work but this isn't performance sensitive code so who cares 👍

Comment (mrgleam) [reply src/docs.rs:109]: Thank you @lpil for your recommend.
I try again in this morning  and it work. I can use` into_iter()` when I remove  reference in `doc_links`

This is ok?


Screen Shot 2564-01-20 at 09 58 28

Comment (lpil) [reply src/docs.rs:109]: That looks great! I think we can also drop the `collect` on `doc_links` and then remove the `into_iter` when it is referenced on line 106

Comment (mrgleam) [reply src/docs.rs]: Thank you @lpil 

It has been done :)

**Review by lpil — APPROVED** [review]
Sweet! Thank you

**Pull request merged**